ext/ddtrace.stub.php

```diff
@@ -0,0 +1,496 @@
+
+    class SpanStack {
+        /**
+         * @var SpanStack|null The parent stack, or 'null' if there is none
+         */
+        public SpanStack|null $parent;
+
+        /**
+         * @var SpanData The active span
+         */
+        public SpanData $active;
+    }
+
+    /**
+     * Close child spans of a parent span if a non-internal span is given,
+     * else if 'null' is given, close active, non-internal spans
+     * @param SpanData|null $span The parent span
+     * @return bool|int 'false' if spans couldn't be closed, else the number of span closed
+     */
+    function close_spans_until(?SpanData $span): bool|int {}
+
+    /**
+     * Get the active span
+     * @return SpanData|null 'null' if tracing isn't enabled or there isn't any active span, else the active span
+     */
+    function active_span(): null|SpanData {}
+
+    /**
+     * Get the root span
+     * @return SpanData|null 'null' if tracing isn't enabled or if the active stack doesn't have a root span,
+     * else the root span of the active stack
+     */
+    function root_span(): null|SpanData {}
+
+    /**
+     * Start a new custom user-span on the top of the stack
+     * @param \DateTimeInterface|int|float $startTime Start time of the span in seconds. Defaults to zero.
+     * @return SpanData The newly started span
+     */
+    function start_span(\DateTimeInterface|int|float $startTime = 0): SpanData {}
+
+    /**
+     * Close the currently active user-span on the top of the stack
+     * @param \DateTimeInterface|int|float|null $finishTime Finish time in seconds. if passing float or int, it should
+     * represent the timestamp (including as many decimal places as you need)
+     * @return void
+     */
+    function close_span(\DateTimeInterface|int|float|null $finishTime): void {}
+
+    /**
+     * Start a new trace
+     *
+     * More precisely, a new root span stack will be created and switched on to, and a new span started.
+     *
+     * @return SpanData The newly created root span
+     */
+    function start_trace_span(): SpanData {}
+
+    /**
+     * Get the active stack
+     * @return SpanStack|null A copy of the active stack, or 'null' if there is none
+     */
+    function active_stack(): SpanStack|null {}
+
+    /**
+     * Initialize a new span stack and switch to it. If tracing isn't enabled, a root span stack will be created.
+     * @return SpanStack The newly created span stack
+     */
+    function create_stack(): SpanStack {}
+
+    /**
+     * Switch back to a specific stack (even if there is no active span on that stack), or to the parent of the active
+     * stack if no stack is given.
+     * @param SpanData|SpanStack|null $newStack Stack to switch to. If 'null' is given, switches to the parent of the
+     * active stack. If a SpanData object is given, it will switch to the stack of the latter.
+     * @return SpanStack|null The newly active stack, or 'null' if there wasn't an active stack and no specific stack
```

Comment:
Like with active_stack, It's really, or 'null' if the tracer is disabled. Won't happen under normal operation.